compiler/ballerina-lang/src/main/java/io/ballerina/projects/Project.java

```diff
@@ -82,6 +84,13 @@ public Path sourceRoot() {
     }
 
     public Path targetDir() {
+        if (this.projectKind == ProjectKind.SINGLE_FILE_PROJECT) {
+            try {
+                return Files.createTempDirectory("ballerina-cache" + System.nanoTime());
```

Comment:
This will create temp directories each time we call the function. We can improve this as follows: 
1. Make `Project.target` abstract and implement it in SingleFileProject and BuildProjects. BalaProject should throw an unsupported operation exception.
2. In SingleFileProject, declare a class level variable (e.g. targetDir) and create and assign the temp location at the project creation
3. BuildProject can contain the existing implementation

Reply:
After this change, shall we try to refactor the tasks in CLI to use this method?